api/src/test/java/com/exelaration/abstractmemery/services/ImageServiceTest.java

```diff
@@ -0,0 +1,74 @@
+package com.exelaration.abstractmemery.services;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.verify;
+
+import com.exelaration.abstractmemery.domains.Image;
+import com.exelaration.abstractmemery.services.implementations.ImageServiceImpl;
+import java.io.IOException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.springframework.mock.web.MockMultipartFile;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+@ExtendWith(SpringExtension.class)
+public class ImageServiceTest {
+
+  @Mock private FileStorageService fileStorageService;
+
+  @Mock private MetadataService metadataService;
+
+  @InjectMocks private ImageServiceImpl imageService;
+
+  @BeforeEach
+  public void init() {
+    MockitoAnnotations.initMocks(this);
+  }
+
+  @Test
+  public void imageServiceSave_WhenImageSavedSuccessfully_ExpectImageReturned() throws IOException {
```

Comment:
These shouldn't need the `throws` anymore because it will no longer throw an Exception.  We're now handling the exceptions in the `ImageService` and it won't hit this level.